### PR TITLE
Remove support for Python 3.4

### PR DESCRIPTION
**Pull request by jd**
Python 3.4 has reached end of life is not supported anymore.

### CODE REVIEW

**Comment by jd** [comment]
@brettlangdon …because? (genuinely asking)

**Comment by brettlangdon** [comment]
We still have a few (albeit low number) of customers still running `3.4.x`, we should figure out a migration and support plan for them.

**Comment by jd** [comment]
They have to upgrade to a more recent Python anyhow. I imagine they can stick to the latest version that supports Python 3.4 of `ddtrace`.
Many of the integration we support have dropped or will drop Python 3.4 too anyhow 🤷‍♂

File: setup.py [hunk 1]
```diff
@@ -123,7 +123,6 @@ def run_tests(self):
     classifiers=[
         "Programming Language :: Python",
         "Programming Language :: Python :: 2.7",
-        "Programming Language :: Python :: 3.4",
```
Comment (brettlangdon) [inline setup.py]: Should we add `python_requires`?

Comment (jd) [reply setup.py]: I don't think it can hurt. I don't know if `pip` implements it but it's a good idea anyhow.

**Comment by mrmachine** [comment]
@brettlangdon @jd Please do not arbitrarily make packages un-installable on older (EOL) versions of Python *JUST* because Python itself is EOL and not because the package has any inherent incompatibility with that version of Python.

I am working on upgrading an old project and I need to go to Python 3 (specifically for `ddtrace` profiling) but other software in my stack requires Python<=3.3. This is only a stepping stone upgrade, but it'll be a big job to upgrade those other dependencies to more current (not EOL) versions of Python 3.

I can't even attempt to *install* the package, and I have no idea why or if it is actually incompatible or not. Only this PR which makes it incompatible because Python is EOL.

By all means, do not actively *support* your package on EOL versions of Python. If there are any compatibility issues, close them as "wontfix" and leave it up to them to fork and update if necessary. But if you have no knowledge of *actual* incompatibility, beyond Python being EOL, at least allow me to install the package to find out for myself and keep using it unsupported.

**Pull request merged**